### PR TITLE
Fix Application Inheritance

### LINKED ISSUE #25: Application inheritance does not work
### Steps to reproduce

Define a new application, something like the following:

```
from ramble.appkit import *
from ramble.app.builtin.hostname import Hostname as BaseHostname

class NewHostname(BaseHostname):
  workload('foo', executable='serial')
```

And the `NewHostname` class does not contain any of the attributes defined in the `BaseHostname` class.

### Error message

_No response_

### Information on your system

```
└─▪ ramble debug report
* **Ramble:** 0.0.1 (09f8310125241715016f774b85dcab5057e88f00)
* **Python:** 3.10.8
* **Platform:** linux-debianrodete-skylake
```

### General information

- [X] I have run `ramble debug report` and reported the version of Ramble/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

### PR DESCRIPTION
**Pull request by douglasjacobsen**
This merge updates the DirectiveMeta class to be more in-line with spack v0.19.0's. Performing this update fixes application inheritance.

This merge also adds tests to help ensure application inheritance continues to work in the future.

Fixes #25

### CODE REVIEW

File: lib/ramble/ramble/language/language_base.py [hunk 1]
```diff
@@ -63,65 +63,39 @@ def __new__(cls, name, bases, attr_dict):
...
             except AttributeError:
                 # The base class didn't have the required attribute.
                 # Continue searching
                 pass
 
         # De-duplicates directives from base classes
-        attr_dict['_directives_to_be_executed'] = [
-            x for x in llnl.util.lang.dedupe(
-                attr_dict['_directives_to_be_executed'])]
+        attr_dict["_directives_to_be_executed"] = [
```
Comment (rfbgo) [inline lib/ramble/ramble/language/language_base.py:77]: tiny nit: why are you swapping out ' for " ?

**Review by rfbgo — APPROVED** [review]
LGTM

File: lib/ramble/ramble/language/language_base.py (hunk 1 above)
Comment (douglasjacobsen) [reply lib/ramble/ramble/language/language_base.py:77]: No real reason. I'm just copying spack's directive meta class, so we stay aligned: https://github.com/spack/spack/blob/develop/lib/spack/spack/directives.py#L148

Comment (rfbgo) [reply lib/ramble/ramble/language/language_base.py:77]: OK! I wonder if we should decide what style we want to follow..

**Review by rfbgo — APPROVED** [review]
LGTM

**Pull request merged**